py_v_sdk/contract/atomic_swap_ctrt.py

```diff
@@ -162,3 +184,230 @@ async def get_tok_bal(self, addr: str) -> int:
+        """
+        maker_lock locks the token by the maker.
+        Args:
+            by (acnt.Account): The action taker.
+            amount (Union[int, float]): The amount of the token to be locked.
+            recipient (str): The taker's address.
+            secret (str): The secret.
+            expire_time (int): The expired timestamp to lock.
+            attachment (str, optional): [description]. Defaults to "".
+            fee (int, optional): [description]. Defaults to md.ExecCtrtFee.DEFAULT.
+
+        Returns:
+            Dict[str, Any]: The response returned by the Node API.
+        """
+        SCALE = 1_000_000_000
+        puzzle_bytes = sha256_hash(secret.encode("latin-1"))
+
+        tok_id = await self.token_id
+        resp = await by.chain.api.ctrt.get_tok_info(tok_id)
+        unit = resp["unity"]
+
+        data = await by._execute_contract(
+            tx.ExecCtrtFuncTxReq(
+                ctrt_id=self._ctrt_id,
+                func_id=self.FuncIdx.LOCK,
+                data_stack=de.DataStack(
+                    de.Amount.for_tok_amount(amount, unit),
+                    de.Addr(md.Addr(recipient)),
+                    de.Bytes(md.Bytes(puzzle_bytes)),
+                    de.Timestamp(md.VSYSTimestamp(int(expire_time * SCALE))),
+                ),
+                timestamp=md.VSYSTimestamp.now(),
+                attachment=md.Str(attachment),
+                fee=md.ExecCtrtFee(fee),
+            )
+        )
+        logger.debug(data)
+        return data
+
+    async def taker_lock(
+        self,
+        by: acnt.Account,
+        amount: Union[int, float],
+        maker_swap_ctrt_id: str,
+        recipient: str,
+        maker_lock_tx_id: str,
+        expire_time: int,
+        attachment: str = "",
+        fee: int = md.ExecCtrtFee.DEFAULT,
+    ) -> Dict[str, Any]:
+        """
+        taker_lock locks the token by the maker.
+
+        Args:
+            by (acnt.Account): The action maker.
+            amount (Union[int, float]): The amount of the token to be locked.
+            maker_swap_ctrt_id: The contract id of the maker's.
+            recipient (str): The maker's address.
+            maker_lock_tx_id (str): The tx id of the maker's.
+            expire_time (int): The expire timestamp to lock.
+            attachment (str, optional): [description]. Defaults to "".
+            fee (int, optional): [description]. Defaults to md.ExecCtrtFee.DEFAULT.
+
+        Returns:
+            Dict[str, Any]: The response returned by the Node API.
+        """
+        SCALE = 1_000_000_000
+
+        puzzle_db_key = self.DBKey.for_puzzle(maker_lock_tx_id)
+        print(puzzle_db_key.b58_str)
+        data = await self.chain.api.ctrt.get_ctrt_data(
+            maker_swap_ctrt_id, puzzle_db_key.b58_str
+        )
+        logger.debug(data)
+        hashed_secret_b58str = data["value"]
+        puzzle_bytes = base58.b58decode(hashed_secret_b58str.encode("latin-1"))
```

Comment:
```suggestion
        puzzle_bytes = base58.b58decode(hashed_secret_b58str)
```

`base58.b58decode` can handle string directly.